client/pkg/template/internal/models/models.go

```diff
@@ -30,6 +34,35 @@ type TranslateContext struct {
 	ClusterName string
 }
 
+// SystemExtensions is embedded in Cluster, MachineSet and Machine objects.
+type SystemExtensions struct {
+	SystemExtensions []string `yaml:"systemExtensions,omitempty"`
+}
+
+func (s *SystemExtensions) translate(ctx TranslateContext, nameSuffix string, labels ...pair.Pair[string, string]) []resource.Resource {
+	if len(s.SystemExtensions) == 0 {
+		return nil
+	}
+
+	slices.Sort(s.SystemExtensions)
```

Comment:
I'm not sure if we should do this (anywhere) - different order produces a different output (as the extensions are layered on top of each other). So I think it makes sense to not do this anywhere, or do it everywhere.

IMO not doing makes a bit more sense as the order actually matters. In the frontend, we list them in an ordered manner anyway, and for the templates, we can communicate the fact that order matters in the docs or so.

Reply:
That's a great question. I would rather expect it to be order independent.

Reply:
OK, gotcha.